qutip/quantum_systems/quantum_system.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import qutip as qt
+from typing import Dict, List
+
+
+class QuantumSystem:
+    """
+    General class for quantum systems
+
+    All quantum systems are instances of this class, configured by factory functions.
+    """
+
+    def __init__(self, name: str, **kwargs):
+        """
+        Initialize quantum system
+
+        Parameters:
+        -----------
+        name : str
+            Name/type of the quantum system
+        **kwargs : dict
+            System-specific parameters
+        """
+        self.name = name
+        self.parameters = kwargs
+
+        self.operators = {}
+        self.hamiltonian = None
+        self.c_ops = []
+        self.latex = ""
+
+    def get_operators(self) -> Dict:
+        """Get operators dictionary"""
+        return self.operators
+
+    def get_hamiltonian(self) -> qt.Qobj:
+        """Get Hamiltonian"""
+        return self.hamiltonian
+
+    def get_c_ops(self) -> List[qt.Qobj]:
+        """Get collapse operators"""
+        return self.c_ops
+
+    def get_latex(self) -> str:
+        """Get LaTeX representation"""
+        return self.latex
+
+    @property
+    def dimension(self) -> int:
+        """Get Hilbert space dimension"""
+        return self.hamiltonian.shape[0] if self.hamiltonian else 0
+
+    @property
+    def eigenvalues(self) -> np.ndarray:
+        """Get eigenvalues of the Hamiltonian"""
+        return self.hamiltonian.eigenenergies()
+
+    @property
+    def eigenstates(self) -> tuple:
+        """Get eigenvalues and eigenstates"""
+        return self.hamiltonian.eigenstates()
```

Comment:
just to keep track, i think eric suggested cacheing or storing these in a property?